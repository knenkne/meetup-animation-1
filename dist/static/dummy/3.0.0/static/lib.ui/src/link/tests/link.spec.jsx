import React from 'react'
import { shallow, mount } from 'enzyme'

import { Link } from '../link'
import { Icon } from '../../icon'

describe('<Link />', () => {
    it('to be available', () => {
        expect(Link).toBeDefined()
    })

    it('render internal link with mode', () => {
        const wrapper = shallow(<Link external={false} href="#" />)
        expect(wrapper.find(Icon).length).toBe(0)

        const wrapper2 = shallow(<Link external={false} href="#" mode="file:word" />)
        expect(wrapper2.find(Icon).prop('name')).toBe('icon:core/common/file:word')
    })

    it('render external link', () => {
        const wrapper = shallow(<Link external href="https://" />)
        expect(wrapper.find('a').first().prop('target')).toBe('_blank')
        expect(wrapper.find('a').first().prop('rel')).toBe('noopener noreferrer')
        expect(wrapper.find(Icon).first().prop('name')).toBe('icon:core/common/external')
    })

    it('render internal link with mode', () => {
        const wrapper = shallow(<Link external={false} href="/someone" />)
        expect(wrapper.find('a').prop('target')).not.toBe('_blank')
        expect(wrapper.find('a').prop('rel')).not.toBe('noopener noreferrer')
        expect(wrapper.find('a').prop('href').indexOf('/someone')).toBeGreaterThan(-1)
        expect(wrapper.find(Icon).length).toBe(0)

        const wrapper2 = shallow(<Link external={false} href="/someone" mode="file:word" />)
        expect(wrapper2.find(Icon).first().prop('name')).toBe('icon:core/common/file:word')
    })

    it('рендерит внутреннюю ссылку по хэшу', () => {
        const wrapper = shallow(<Link external={false} href="#hello" />)
        expect(wrapper.find('a').prop('href').indexOf('#hello')).toBeGreaterThan(-1)
    })

    it('рендерит bold ссылку', () => {
        const wrapper = shallow(<Link bold href="#" />)
        expect(wrapper.hasClass(Link.theme.linkBold)).toBeTruthy()
    })

    it('рендерит прстую икону external', () => {
        const wrapper = shallow(<Link external href="#" />)
        expect(wrapper.find(Icon).prop('name')).toBe('icon:core/common/external')
    })

    it('рендерит bold икону external', () => {
        const wrapper = shallow(<Link external bold href="#" />)
        expect(wrapper.find(Icon).prop('name')).toBe('icon:core/common/externalBold')
    })
    it('создает обертку вокруг иконки и последнего слова во внешней ссылке', () => {
        const component = mount(<Link external href="#">{'Внешняя ссылка'}</Link>)

        expect(component).toMatchSnapshot()
    })

    it('создает обертку вокруг иконки и единственного слова во внешней ссылке', () => {
        const component = mount(<Link external href="#">{'Cсылка'}</Link>)

        expect(component).toMatchSnapshot()
    })
})
