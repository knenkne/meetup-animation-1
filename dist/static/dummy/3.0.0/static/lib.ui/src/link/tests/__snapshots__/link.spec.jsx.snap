// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> создает обертку вокруг иконки и единственного слова во внешней ссылке 1`] = `
<Link
  as="a"
  bold={false}
  colorScheme="green"
  external={true}
  href="#"
  mode=""
  size="lg"
  theme={null}
>
  <SimpleExternalLink
    className="link lg green"
    data-unit="link:external"
    href="#"
    iconName="icon:core/common/external"
    rel="noopener noreferrer"
    target="_blank"
    theme={null}
  >
    <a
      className="link lg green"
      data-unit="link:external"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="externalWrapper"
      >
        <span
          className="last"
        >
          Cсылка
        </span>
        <Icon
          name="icon:core/common/external"
          onClick={[Function]}
          onError={[Function]}
          size="self"
          theme={
            Object {
              "self": "self icon right",
            }
          }
        >
          <span
            className="self icon right"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path fill-rule=\\"nonzero\\" d=\\"M11 8H8.5a.5.5 0 00-.5.5v7a.5.5 0 00.5.5h7a.5.5 0 00.5-.5V13h1v3a1 1 0 01-1 1H8a1 1 0 01-1-1V8a1 1 0 011-1h3v1zm5.793-1.5H14a.5.5 0 110-1h4a.498.498 0 01.5.5v4a.5.5 0 11-1 0V7.207l-5.146 5.147a.5.5 0 01-.708-.708L16.793 6.5z\\"/></svg>",
              }
            }
            data-unit="icon"
            onClick={[Function]}
            onError={[Function]}
          />
        </Icon>
      </span>
    </a>
  </SimpleExternalLink>
</Link>
`;

exports[`<Link /> создает обертку вокруг иконки и последнего слова во внешней ссылке 1`] = `
<Link
  as="a"
  bold={false}
  colorScheme="green"
  external={true}
  href="#"
  mode=""
  size="lg"
  theme={null}
>
  <SimpleExternalLink
    className="link lg green"
    data-unit="link:external"
    href="#"
    iconName="icon:core/common/external"
    rel="noopener noreferrer"
    target="_blank"
    theme={null}
  >
    <a
      className="link lg green"
      data-unit="link:external"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="content"
      >
        Внешняя 
      </span>
      <span
        className="externalWrapper"
      >
        <span
          className="last"
        >
          ссылка
        </span>
        <Icon
          name="icon:core/common/external"
          onClick={[Function]}
          onError={[Function]}
          size="self"
          theme={
            Object {
              "self": "self icon right",
            }
          }
        >
          <span
            className="self icon right"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path fill-rule=\\"nonzero\\" d=\\"M11 8H8.5a.5.5 0 00-.5.5v7a.5.5 0 00.5.5h7a.5.5 0 00.5-.5V13h1v3a1 1 0 01-1 1H8a1 1 0 01-1-1V8a1 1 0 011-1h3v1zm5.793-1.5H14a.5.5 0 110-1h4a.498.498 0 01.5.5v4a.5.5 0 11-1 0V7.207l-5.146 5.147a.5.5 0 01-.708-.708L16.793 6.5z\\"/></svg>",
              }
            }
            data-unit="icon"
            onClick={[Function]}
            onError={[Function]}
          />
        </Icon>
      </span>
    </a>
  </SimpleExternalLink>
</Link>
`;
